src/Lumper.Lib/BSP/IO/IoHandler.cs

```diff
@@ -0,0 +1,51 @@
+namespace Lumper.Lib.BSP.IO;
+
+using System;
+using System.Threading;
+
+/// <summary>
+/// Utility class that can be passed around during save/load to handle progress and cancellation.
+///
+/// Not being overly fancy with progress, doesn't need to be perfectly accurate.
+/// Important thing is that the user is shown ongoing progress as slow operations
+/// like pakfile compression are ongoing.
+/// </summary>
+public class IoHandler(CancellationTokenSource cts)
```

Comment:
the IoHandler appears to not handle Io .. but I'm bad with names so I don't know what to call it

Reply:
It's used in functions handling disk and network, and it *does* handle cancellation... so idk, maybe ProgressHandler / ProgressManager? Eh